pageserver/src/basebackup.rs

```diff
@@ -170,18 +147,41 @@ impl<'a> Basebackup<'a> {
 
     //
     // Extract pg_filenode.map files from repository
+    // Along with them also send PG_VERSION for each database.
     //
     fn add_relmap_file(&mut self, tag: &ObjectTag, db: &DatabaseTag) -> anyhow::Result<()> {
         let img = self.timeline.get_page_at_lsn_nowait(*tag, self.lsn)?;
-        info!("add_relmap_file {:?}", db);
         let path = if db.spcnode == pg_constants::GLOBALTABLESPACE_OID {
+            let dst_path = format!("PG_VERSION");
+            //TODO fix hardcoded value. Get this version num somewhere
+            let data = "14".as_bytes();
+            let header = new_tar_header(&dst_path, data.len() as u64)?;
+            self.ar.append(&header, &data[..])?;
+
+            let dst_path = format!("global/PG_VERSION");
+            let data = "14".as_bytes();
+            let header = new_tar_header(&dst_path, data.len() as u64)?;
+            self.ar.append(&header, &data[..])?;
+
             String::from("global/pg_filenode.map") // filenode map for global tablespace
         } else {
             // User defined tablespaces are not supported
             assert!(db.spcnode == pg_constants::DEFAULTTABLESPACE_OID);
-            let src_path = format!("{}/base/1/PG_VERSION", self.snappath);
+
+            // Append dir path for each database
+            let path = format!("base/{}", db.dbnode);
+
+            // FIXME It's a hack to send dir using append_dir()
```

Comment:
Is there more straightforward way to do it?
https://stackoverflow.com/questions/12844384/add-empty-directory-to-tarfile

Reply:
Sadly this link is about python)